Gnomad tests

tiboloic opened the pull request:
add command to upload required files to GCP bucket (transcripts.json)

tiboloic committed:
Create gnomadv4_extract_test.py

Scripts to extract variants for a panel of 10 genes from gnomAD v4
and save as a hail Table (rows only)

tiboloic committed:
Moved transcript list to json file

Moved the set of transcripts into a json file
Updated code to load transcripts from file

tiboloic committed:
get ready for analysis-runner

Added main.py to encapsulate script
Added safe guard to export mt
Changed GENCODE version to v39
Used output_path() to get path right
Corrected a transcript ID

tiboloic committed:
setup for analysis runner

Added comments
Added a filtering to remove REF pseudo variants
Added setup of dataproc (parameters) in main.py

tiboloic committed:
nits on main.py before first run

tiboloic committed:
nits

replaced " by '
persisted ht to avoid double evaluation

tiboloic committed:
nits

get GNECODE v39 from CPG reference bucket
change the name of the ht to small.ht

tiboloic committed:
add gsutil command to copy needed files

lgruen commented on gnomad/main.py:
```diff
@@ -0,0 +1,41 @@
+cluster.add_job('gnomadv4_extract_test.py', job_name='first_v4_extract_test')
+
+# add the needed files
+import command 
+
+res = command.run(['gsutil cp transcripts.json gs://cpg-constraint-main/transcripts.json']) 
+print(res.output)
```
The standard way to do this is `subprocess.run` with `check=True` set, using a list instead of shell execution:
https://docs.python.org/3/library/subprocess.html#using-the-subprocess-module

Module imports should typically be at the top of the file.

lgruen replied:
Instead of putting this in the root folder of the bucket, it would be better to use `output_path` here as well.

tiboloic replied:
implemented

tiboloic committed:
Merge remote-tracking branch 'origin/main' into gnomad_tests

tiboloic committed:
use subprocess to run file copy

tiboloic committed:
output_path

main.py: added output_path() to write needed files to correct location
gnomadv4_extract_test.py: added output_path() to find the location of the transcripts file

lgruen commented on gnomad/gnomadv4_extract_test.py:
```diff
@@ -14,7 +14,7 @@
 # load list of transcripts
 # a panel of 10 genes selected for initial analysis by the structural-constraint project
-transcripts_file = open('transcripts.json', 'r')
+transcripts_file = open(output_path('transcripts.json'), 'r')
```
Python can't directly read from GCS, but we're using `cloudpathlib`'s `AnyPath` in many places.

tiboloic replied:
OK that is becoming way too much friction, can we please run this analysis outside of the analysis runner ?

lgruen replied:
This has nothing to do with the analysis-runner. If you submitted this script from your laptop to a Dataproc cluster, you'd run into exactly the same problem.

Also this is a two line change, I wouldn't quite call this "way too much friction"...

```py
from cloudpathlib import AnyPath

with AnyPath(output_path('transcripts.json')).open() as transcripts_file:
    TRANSCRIPTS = json.load(transcripts_file)
```

tiboloic replied:
Thanks Leo, that's helpful

lgruen commented on gnomad/main.py:
```diff
@@ -31,6 +32,9 @@
 cluster.add_job('gnomadv4_extract_test.py', job_name='first_v4_extract_test')
 
+# add the needed files
+subprocess.run(['gsutil cp transcripts.json', output_path('transcripts.json')]) 
```
```suggestion
subprocess.run(['gsutil', 'cp', 'transcripts.json', output_path('transcripts.json')], check=True)
```

tiboloic committed:
Update gnomad/main.py

Co-authored-by: Leonhard Gruenschloss <leonhard@gruenschloss.org>

tiboloic committed:
Update gnomadv4_extract_test.py

use AnyPath to open file located on GPG bucket
